cluster/aws/util.sh

```diff
@@ -721,14 +723,11 @@ function create-iam-profile {
 
 # Creates the IAM roles (if they do not already exist)
 function ensure-iam-profiles {
-  aws iam get-instance-profile --instance-profile-name ${IAM_PROFILE_MASTER} || {
```

Comment:
> examples being: Federated Services, Federated Replica Sets, Kubernetes Deployments and in various other places.

All these examples are within the e2e framework? This file deals solely with provisioning AWS resources, not Kubernetes api objects. If you look at any other type of AWS resource which is created in `kube-up.sh`, it's also destroyed in `kube-down.sh`.

If we don't do this, then any changes made to the IAM roles/instance profiles will not be propagate properly to e2e tests. The IAM roles/profiles will have to be managed manually.

Finally, why are we so concerned about deleted these IAM roles? They're created and used exclusively by the cluster (just like all the other AWS resources), so I don't know why we're so concerned about retaining them. These IAM objects exist simply to allow the ec2 instances to access the AWS API- can you state explicitly the rationale behind not deleting them when the cluster is torn down?